pythonFiles/include/blender_vscode/operators/addon_update.py

```diff
@@ -19,8 +19,15 @@ def execute(self, context):
             send_dict_as_json({"type" : "disableFailure"})
             return {'CANCELLED'}
 
-        for name in list(sys.modules.keys()):
-            if name.startswith(self.module_name):
+        root = None
+        for (name, module) in sys.modules.copy().items():
+            if not hasattr(module, '__file__'):
+                continue
+
+            if name == self.module_name and hasattr(module, '__path__'):
+                root = module.__path__[0] + '\\'
```

Comment:
File paths should not be handled like this. This is not cross platform compatible. Please use either `os.path` or `pathlib` when dealing with file paths.

Reply:
`root` is the folder of the addon.
It reads the modules in `sys` to find the first module containing in the addon's folder  _(I assume `__init__.py` is always first but I didn't/don't care to verify that)_.
Until a folder is found, it will continue to the end of the modules.

When the `root` is found, modules afterwards will only check if they're in a subdirectory of the `root` folder.
If they are in the folder, then delete the module.

- - - - - 

You are free to treat this as a bug report or whatever, and redo it anyway you want. 
I use Windows and only care that it works for me.
Just uploaded for others, to save time debugging what went wrong